datalad_registry/make_celery.py

```diff
@@ -1,9 +1,60 @@
 # This file provides Celery commands access to the Celery app created through
 # the factory functions in datalad_registry/__init__.py
+import logging
+import re
 
 from celery import Celery
 
 from . import create_app
 
+
+# === Code for suppressing known git progress reports ===
+class SuppressKnownGitProgressReport(logging.Filter):
+    # Known git progress report types
+    # These types can be found in the definition of
+    # `datalad.support.gitrepo.GitProgress`
+    known_git_progress_report_types = {
+        "Counting objects",
+        "Compressing objects",
+        "Writing objects",
+        "Receiving objects",
+        "Resolving deltas",
+        "Finding sources",
+        "Checking out files",
+        "Enumerating objects",
+    }
+
+    re_op_absolute = re.compile(r"(?:remote: )?([\w\s]+):\s+\d+.*")
+    re_op_relative = re.compile(r"(?:remote: )?([\w\s]+):\s+\d+% \(\d+/\d+\).*")
+
+    def filter(self, record):
+        # The following logic is based on the logic in
+        # `datalad.support.gitrepo.GitProgress._parse_progress_line`
+
+        msg = record.getMessage()
+
+        match = self.re_op_relative.match(msg)
+        if match is None:
+            match = self.re_op_absolute.match(msg)
+
+        if match is None:
+            # === msg does not match the pattern of a git progress report ===
+            return True
+
+        op_name = match.group(1)
+
+        # Return False (filtering out the log message) only
+        # if the message matches the pattern of a git progress report and
+        # is of a known git progress report type
+        return op_name not in self.known_git_progress_report_types
+
+
+# Retrieve a reference to the "datalad.gitrepo" logger
+dl_gitrepo_lgr = logging.getLogger("datalad.gitrepo")
+
+# Add a filter to the "datalad.gitrepo" logger to suppress known git progress reports
+dl_gitrepo_lgr.addFilter(SuppressKnownGitProgressReport())
```

Comment:
This isn't need any longer right?

Reply:
Currently, it is still needed. I added the solution in your previous comment. However, it only removed the duplications of the "start messages" such as "Start counting objects".

I also discovered that the log level setting, https://github.com/datalad/datalad-registry/blob/523963d7377e11df9d0acaa908bc380ef0962850/docker-compose.dev.yml#L41, of the logger seemingly has not affect on what messages are logged in execution.

This PR requires further digging. I will come back to this PR after #252 which is more important.

Reply:
ok, so here, I guess, now we are having `log_progress` still causing logging of the progress... complimentary issue I guess.  I guess we need to  dig deeper, most likely adding a way to completely disable log_progress. It might be that some workaround like

```python
def no_log_progress(*args, **kwargs):
     pass

from datalad.support import gitrepo
gitrepo.log_progress = no_log_progress
```

would do the drill.

Reply:
This solution would take away the logs we want to take away and more. I think I still have to "start" and "finishing" logs, for example, to remain.

Reply:
Additionally,  I still want an explanation for why changing the log level of the Celery worker command doesn't change the behavior of the worker's logger.

Reply:
I think it is again datalad to blame: I added a note (see "edit 1") to https://github.com/datalad/datalad/pull/7521

Reply:
Michael also pointed to the fact that we might want to use `.propagate` to avoid propagation of messages from our datalad logger: https://docs.python.org/3/library/logging.html#logging.Logger